addon/components/polaris-text-field.js

```diff
@@ -392,79 +389,82 @@ export default Component.extend({
     }
 
     return describedBy.join(' ');
-  }),
+  }).readOnly(),
 
   ariaLabelledBy: computed('id', function() {
     return `${ this.get('id') }Label`;
-  }),
+  }).readOnly(),
 
-  ariaInvalid: computed('error', function() {
-    return !!this.get('error');
-  }),
+  ariaInvalid: bool('error').readOnly(),
 
   inputType: computed('type', function() {
     let type = this.get('type');
 
     return type === 'currency' ? 'text' : type;
-  }),
-
-  input: computed('multiline', function() {
-    let multiline = this.get('multiline');
-
-    return multiline ? 'textarea' : 'input';
-  }),
+  }).readOnly(),
 
   minimumLines: computed('multiline', function() {
     let multiline = this.get('multiline');
 
-    return (typeof multiline === 'number') ? multiline : 1;
-  }),
+    return (typeOf(multiline) === 'number') ? multiline : 1;
+  }).readOnly(),
 
   heightStyle: computed('height', 'multiline', function() {
     let { height, multiline } = this.getProperties('height', 'multiline');
 
-    return (height && multiline) ? htmlSafe(`height: ${ height }px`) : '';
-  }),
+    return (height && multiline) ? htmlSafe(`height: ${ height }px`) : null;
+  }).readOnly(),
+
+  shouldShowSpinner: computed('disabled', 'type', function() {
+    let { type, disabled } = this.getProperties('type', 'disabled');
+
+    return (typeOf(type) === 'number') && !disabled;
+  }).readOnly(),
+
+  addValueListener() {
+    this.addEventListener('keyup', () => {
+      this.debounceTask('debouncedUpdateValue', 250);
+    });
+  },
+
+  debouncedUpdateValue() {
+    let field = this.get('fieldElement');
+    this.set('value', field.value);
+  },
 
   init() {
     this._super(...arguments);
 
     let { id, type } = this.getProperties('id', 'type');
-    let getUniqueID = createUniqueIDFactory('TextField');
 
     assert(`ember-polaris::polaris-text-field - ${ type } is not a valid type.`, allowedTypes.indexOf(type) > -1);
 
-    id = id || getUniqueID();
+    id = id || `TextField-${ guidFor(this) }`;
 
     this.setProperties({
       height: null,
       focus: false,
       id
     });
-
-    next(this, ()=> {
-      this.addValueListener();
-    });
   },
 
-  didReceiveAttrs() {
-    this._super(...arguments);
-
-    let { input, focused } = this.getProperties('input', 'focused');
+  checkFocus() {
+    let { fieldElement, focused } = this.getProperties('fieldElement', 'focused');
 
-    if (input && focused) {
-      input.focus();
+    if (fieldElement && focused) {
+      this.$(fieldElement).focus();
     }
   },
 
-  willDestroyElement() {
-    this.removeValueListener();
+  didInsertElement() {
+    this.set('fieldElement', this.element.querySelector('input, textarea'));
+    this.addValueListener();
+    this.checkFocus();
   },
 
-  destroy() {
-    runDisposables(this);
-
+  didReceiveAttrs() {
     this._super(...arguments);
+    this.checkFocus();
```

Comment:
This might need to be scheduled to use the run loop, since I'm not sure the attributes will have updated by the time this runs (you're probably alright though)

Reply:
Did a quick test, we good -

![focus](https://user-images.githubusercontent.com/2292367/42227350-51af0c8a-7eaf-11e8-837e-b56fcd2b636d.gif)